crates/control/src/motion/head_motion.rs

```diff
@@ -46,67 +44,83 @@ pub struct MainOutputs {
 
 impl HeadMotion {
     pub fn new(_context: CreationContext) -> Result<Self> {
-        Ok(Self {
-            last_request: Default::default(),
-        })
+        Ok(Self {})
     }
 
     pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
-        let raw_request = if *context.has_ground_contact {
-            Self::joints_from_motion(&context)
-        } else {
-            Default::default()
-        };
+        let HeadJointsCommand {
+            positions: raw_positions,
+            stiffnesses,
+        } = context
+            .has_ground_contact
+            .then(|| Self::joints_from_motion(&context))
+            .unwrap_or_else(|| HeadJointsCommand {
+                positions: Default::default(),
+                stiffnesses: HeadJoints::fill(0.8),
+            });
 
         let maximum_movement =
             *context.maximum_velocity * context.cycle_time.last_cycle_duration.as_secs_f32();
 
-        let controlled_request = HeadJoints {
-            yaw: self.last_request.yaw
-                + (raw_request.yaw - self.last_request.yaw)
+        let controlled_positions = HeadJoints {
+            yaw: context.sensor_data.positions.head.yaw
+                + (raw_positions.yaw - context.sensor_data.positions.head.yaw)
                     .clamp(-maximum_movement.yaw, maximum_movement.yaw),
-            pitch: self.last_request.pitch
-                + (raw_request.pitch - self.last_request.pitch)
+            pitch: context.sensor_data.positions.head.pitch
+                + (raw_positions.pitch - context.sensor_data.positions.head.pitch)
                     .clamp(-maximum_movement.pitch, maximum_movement.pitch),
         };
 
-        let pitch_max = if controlled_request.yaw.abs() >= *context.outer_yaw {
+        let maximum_pitch = if controlled_positions.yaw.abs() >= *context.outer_yaw {
             *context.outer_maximum_pitch
         } else {
             let interpolation_factor =
-                0.5 * (1.0 + (PI / *context.outer_yaw * controlled_request.yaw).cos());
+                0.5 * (1.0 + (PI / *context.outer_yaw * controlled_positions.yaw).cos());
             *context.outer_maximum_pitch
                 + interpolation_factor
                     * (*context.inner_maximum_pitch - *context.outer_maximum_pitch)
         };
-        let clamped_pitch = controlled_request.pitch.clamp(0.0, pitch_max);
 
-        let clamped_request = HeadJoints {
+        let clamped_pitch = controlled_positions.pitch.clamp(0.0, maximum_pitch);
+        let clamped_positions = HeadJoints {
             pitch: clamped_pitch,
-            yaw: controlled_request.yaw,
+            yaw: controlled_positions.yaw,
         };
 
-        self.last_request = controlled_request;
```

Comment:
@schmidma What was the reason to implement the `HeadMotion` open-loop with a state variable instead of using the current sensor positions? See https://github.com/HULKs/nao/commit/895151d0104a06e7535d6d681c1ffca4f5710a72#diff-e20fc1836385ae33c49171b884fa87cc2036a451dd690dae352724c517b2261eR70

Reply:
Readded open-loop interpolation (via GitHub on my Smartphone in the train 😎)